File: Web/src/app/model/Comment.ts
import { User } from "../model";
export interface Comment {
    commentId:number;
    commentContext:string;
    rating:number;
    datePost:string;
    type:CommentType;
    likeCounter:number;
    user:User;
    userId:number;
    courseId:number;
    lessonId:number;
}
export interface SubComment {
    subCommentId:number;
    subCommentContext:string;
    subDatePost:string;
    likeCounter:number;
    parentCommentId:number;
    user:User;
    userId:number;
}
export enum CommentType
{
    Comment,
    Rating
}

File: BackendService/BackendService/Models/Comment.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace BackendService.Models
{
    public class Comment
    {
        [Key]
        public int CommentId { get; set; }
        public string CommentContext { get; set; }
        public float Rating { get; set; }
        public DateTime DatePost { get; set; }
        public CommentType Type { get; set; }
        public int LikeCounter { get; set; }
        public ICollection<SubComment> SubComments { get; set; }
        public User User { get; set; }
        [Required]
        public int UserId { get; set; }
        public Course Course { get; set; }
        public int? CourseId { get; set; }
        public string? LessonId { get; set; }
    }
    public enum CommentType
    {
        Comment,
        Rating
    }
}
